ffi/derive/src/convert.rs

```diff
@@ -85,19 +79,26 @@ fn variant_discriminants(enum_: &syn::DataEnum) -> Vec<syn::Expr> {
     })
 }
 
-fn derive_try_from_repr_c_for_opaque_item_wrapper(name: &Ident) -> TokenStream2 {
+fn derive_try_from_repr_c_for_opaque_item_wrapper(input: &DeriveInput) -> TokenStream2 {
+    let name = &input.ident;
+    let mut generics = input.generics.clone();
+    let lifetime: syn::Lifetime = syn::parse_quote!('itm);
+    let lifetimes = &[lifetime.clone()];
+    let (generic_params, ty_generics, where_clause) =
+        add_bounds_and_split_generics(&mut generics, lifetimes, &[]);
+
```

Comment:
Seeing that we have so many repeating code makes me itch to refactor `input` to be some custom struct instead of `DeriveInput` just like I did in `config`, but I understand that it's definitely outside of scope here.